### PR TITLE
Alternative formatString implementations

### PR DESCRIPTION
**Pull request by goofballLogic**
YMMV

### CODE REVIEW

File: js/app.js [hunk 1]
```diff
@@ -223,3 +223,55 @@ function formatString(str, args) {
...
+
+// VARIANT 4
+// e.g. formatString("Placeholder {0} and {1}", "Value 1", "Value 2");
+function formatString() {
+    return Array.from(arguments).reduce(function(result, value, index) {
+        return result.replace(`{${index}}`, value);
+    });
+}
+
+
```
Comment (YorgunKelebek) [inline js/app.js]: I am not sure how you would pass and use the string to be formatted in these two variants (5/6)

Comment (goofballLogic) [reply js/app.js]: The usage example is shown in the comment above each one. All examples use the same format string syntax that you originally proposed.

**Comment by goofballLogic** [comment]
Ok do the other variants make sense in terms of how they work? This is a useful example of various types of syntax

**Comment by YorgunKelebek** [comment]
> Ok do the other variants make sense in terms of how they work? This is a useful example of various types of syntax

I understand first 2 - use of Array.reduce I am not too sure about, but I will have a good look. Only 3 works for me, others don't and produce the same incorrect result:

formatString("Hello {0} {1}", ["Hakan,", "my man!"]);
=> Hello {0} Hakan,,my man!

**Comment by goofballLogic** [comment]
> > Ok do the other variants make sense in terms of how they work? This is a useful example of various types of syntax
> 
> I understand first 2 - use of Array.reduce I am not too sure about, but I will have a good look. Only 3 works for me, others don't and produce the same incorrect result:
> 
> formatString("Hello {0} {1}", ["Hakan,", "my man!"]);
> => Hello {0} Hakan,,my man!

Does `formatString("Hello {1} {2}", "Hakan,", "my man!");` work?

**Comment by YorgunKelebek** [comment]
> > > Ok do the other variants make sense in terms of how they work? This is a useful example of various types of syntax
> > 
> > 
> > I understand first 2 - use of Array.reduce I am not too sure about, but I will have a good look. Only 3 works for me, others don't and produce the same incorrect result:
> > formatString("Hello {0} {1}", ["Hakan,", "my man!"]);
> > => Hello {0} Hakan,,my man!
> 
> Does `formatString("Hello {1} {2}", "Hakan,", "my man!");` work?

Yes!

**Comment by YorgunKelebek** [comment]
Got all I needed - thanks

**Pull request closed without merging**